TUM Campus App/PlansTableViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  PlansTableViewController.swift
+//  TUM Campus App
+//
+//  Created by Florian Gareis on 16.12.16.
+//  Copyright © 2016 LS1 TUM. All rights reserved.
+//
+
+import UIKit
+import Alamofire
+
+class PlansTableViewController: UITableViewController {
+    
+    // MARK: - Table view data source
+    lazy var plans = Plans()
```

Comment:
So my suggestion is:

In Plans:

```Swift
class Plans {

    static lazy var shared = Plans()

    // Rest of your code
}
```

And in your TableView instead of this plans variable you'd have:
```Swift
class PlansTableViewController: UITableViewController {

    var plans: [Plan] {
        return Plans.shared.plans
    }

}
```

And in your loop you just do:


```Swift
for plan in plans {
    if (plan.type == .pdf) {
        let downloader = PlanDownloader()
        downloader.downloadPlan(urlString: plan.url, withName: plan.fileUrl)
     }
}
```

Or if you're as lazy as me:

```Swift
plans
  .filter { $0.type == .pdf }
  .forEach { PlanDownloader().downloadPlan(urlString: $0.url, withName: $0.fileUrl) }
```